producer/types.go

```diff
@@ -66,8 +76,9 @@ type Producer interface {
 	Init() error
 
 	// Close stops the producer from accepting new requests immediately.
-	// It will also block until all the data buffered in Producer has been consumed.
-	Close()
+	// If the CloseType is WaitForConsumption, then it will block until all the data has been consumed.
+	// If the CloseType is DropEverything, then it will simply drop all the data buffered and return.
+	Close(CloseType)
```

Comment:
NIT: seems like the pattern is to name the arguments in the interface function signatures.